support-frontend/app/controllers/RedemptionController.scala

```diff
@@ -127,21 +142,27 @@ class RedemptionController(
           userHasSub =>
             if (userHasSub)
               Future.successful(redirectToExistingThankYouPage)
-            else
-              tryToShowProcessingPage(redemptionCode)
+            else {
+              identityService.getUser(request.user.minimalUser).value.flatMap{
+                case Left(message) =>
+                  SafeLogger.error(scrub"could not fetch user: $message")
+                  Future.successful(InternalServerError("could not fetch user"))
```

Comment:
they should be correctly logged in by now.  However we could just assume they are not test user at this point, so they can still check out